pkg/middleware/request_metrics.go

```diff
@@ -79,7 +83,17 @@ func RequestMetrics(features featuremgmt.FeatureToggles, cfg *setting.Cfg, promR
 				}
 			}
 
-			labelValues := []string{handler, code, r.Method}
+			source := errutil.SourceServer
+			// HACK: contexthandler.FromContext(r.Context()) doesn't work???
```

Comment:
Yea I run into these kind of problems when I play around with ideas for the middleware. I think its because we forget to update one of the points towrads the request somewhere. 

Having the golang request, then a point on requestContext and macarons context means we are fragile to just updating one of them when doing somehting like `WithContext` 

Kind of why I wanted to use RequestMetaData since its added one the go request as early as possible in the middleware chain :) I'm really worried we make misstake with RequestContext